### PR TITLE
feat!: add deletion_policy in volumes

### LINKED ISSUE #8: snapshot_policy - Attempt to get attribute from null value
Hello,

This module defaults an empty `snapshot_policy` to null. 

From variables.tf:
```hcl
    snapshot_policy = optional(object({
      enabled = optional(bool, false)
      hourly_schedule = optional(object({
        snapshots_to_keep = optional(number)
        minute            = optional(number)
      }), null)

      daily_schedule = optional(object({
        snapshots_to_keep = optional(number)
        minute            = optional(number)
        hour              = optional(number)
      }), null)

      weekly_schedule = optional(object({
        snapshots_to_keep = optional(number)
        minute            = optional(number)
        hour              = optional(number)
        day               = optional(string)
      }), null)
    }))
```

However, there is check on fields of the `snapshot_policy`, resulting in errors if `snapshot_policy` input isn't passed to the module.

In main.tf:
```hcl
  dynamic "snapshot_policy" {
    for_each = each.value.snapshot_policy.enabled ? ["volume_snapshot_policy"] : []
   ...

      dynamic "daily_schedule" {
        for_each = each.value.snapshot_policy.daily_schedule == null ? [] : ["daily_schedule"]
       ...

      dynamic "weekly_schedule" {
        for_each = each.value.snapshot_policy.weekly_schedule == null ? [] : ["weekly_schedule"]
        ...
  }
```

When the module tries to access `each.value.snapshot_policy.<field>`, it fails if `snapshot_policy` isn't passed as input to the module:
```
module "volumes_only" {
   source  = "GoogleCloudPlatform/netapp-volumes/google"  
   version = "~> 0.1"

  project_id = "my-project-id"
  location   = "my-location"

  storage_pool = {
    create_pool = false
    name        = "pool-name"
  }

  storage_volumes = [
    {
      name       = "test-volume-1"
      share_name = "test-volume-1"
      size       = "100"
      protocols  = ["NFSV3"]
      export_policy_rules = {
        test = {
          allowed_clients = "10.0.0.0/24,10.100.0.0/24"
          access_type     = "READ_WRITE"
          nfsv3           = true
          has_root_access = true
        }
      }
    },
  ]
}
```
```
│ Error: Attempt to get attribute from null value
│ 
│   on .terraform/modules/volume.netapp_volumes/main.tf line 57, in resource "google_netapp_volume" "storage_volumes":
│   57:     for_each = each.value.snapshot_policy.enabled ? ["volume_snapshot_policy"] : []
│     ├────────────────
│     │ each.value.snapshot_policy is null
│ 
│ This value is null, so it does not have any attributes.
╵
```

### LINKED ISSUE #10: Add deletion_policy for volumes

### PR DESCRIPTION
**Pull request by imrannayer**
fix #8 
fix #10

### CODE REVIEW

File: examples/simple_example/main.tf [hunk 1]
```diff
@@ -118,20 +123,14 @@ module "netapp_volumes" {
           has_root_access = true
         }
       }
+
```
Comment (g-awmalik) [inline examples/simple_example/main.tf:126]: nit: whitespace

File: main.tf [hunk 2]
```diff
@@ -52,21 +52,22 @@ resource "google_netapp_volume" "storage_volumes" {
   security_style     = lookup(each.value, "security_style", null)
   kerberos_enabled   = lookup(each.value, "kerberos_enabled", null)
   restricted_actions = lookup(each.value, "restricted_actions", null)
+  deletion_policy    = lookup(each.value, "deletion_policy", null)
 
   dynamic "snapshot_policy" {
-    for_each = each.value.snapshot_policy.enabled ? ["volume_snapshot_policy"] : []
+    for_each = lookup(each.value, "snapshot_policy", null) == null ? [] : ["volume_snapshot_policy"]
```
Comment (g-awmalik) [inline main.tf:58]:
```suggestion
    for_each = each.value.snapshot_policy != null ? ["volume_snapshot_policy"] : []
```

File: versions.tf [hunk 3]
```diff
@@ -19,7 +19,7 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = ">= 5.13, < 6"
+      version = ">= 5.15, < 6"
```
Comment (g-awmalik) [inline versions.tf:22]: Should this be a breaking change?

Comment (imrannayer) [reply versions.tf:22]: Changed it to breaking change

File: main.tf [hunk 4]
```diff
@@ -88,7 +89,7 @@ resource "google_netapp_volume" "storage_volumes" {
   }
 
   dynamic "export_policy" {
-    for_each = each.value.export_policy_rules == null ? [] : ["export_policy_rules"]
```
[the remark this answers is not in the document]
Comment (imrannayer) [reply main.tf]: It should. Changed it back.

**Comment by imrannayer** [comment]
@g-awmalik plz review

**Review by g-awmalik — APPROVED** [review]

**Pull request merged**